go/controller/internal/httpserver/handlers/helpers.go

```diff
@@ -144,16 +146,40 @@ func FlattenStructToMap(data interface{}, targetMap map[string]interface{}) {
 	}
 }
 
-func CreateSecret(kubeClient client.Client, name string, namespace string, data map[string]string) (*corev1.Secret, error) {
+func CreateSecret(ctx context.Context, kubeClient client.Client, name string, namespace string, data map[string]string, owner *metav1.OwnerReference) (*corev1.Secret, error) {
 	secret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name,
 			Namespace: namespace,
 		},
+		Type:       corev1.SecretTypeOpaque,
 		StringData: data,
 	}
 
-	if err := kubeClient.Create(context.Background(), secret); err != nil {
+	if owner != nil {
+		secret.OwnerReferences = []metav1.OwnerReference{*owner}
+	}
+
+	if err := kubeClient.Create(ctx, secret); err != nil {
+		if k8serrors.IsAlreadyExists(err) {
+			// If the secret already exists, try to update it
```

Comment:
I'm actually not sure about this. In what case would the secret previously exist that we'd want to add the owner reference? If the secret previously exists, we by definition DO NOT own it.